packages/record-data/addon/-private/graph/index.ts

```diff
@@ -0,0 +1,178 @@
+import Relationships from '../relationships/state/create';
+
+type StableRecordIdentifier = import('@ember-data/store/-private/ts-interfaces/identifier').StableRecordIdentifier;
+type Store = import('@ember-data/store/-private/system/core-store').default;
+type RecordDataStoreWrapper = import('@ember-data/store/-private').RecordDataStoreWrapper;
+type Relationship = import('../relationships/state/relationship').default;
+type JsonApiRelationship = import('@ember-data/store/-private/ts-interfaces/record-data-json-api').JsonApiRelationship;
+type RelationshipDict = import('@ember-data/store/-private/ts-interfaces/utils').ConfidentDict<Relationship>;
+
+const Graphs = new WeakMap<RecordDataStoreWrapper, Graph>();
+
+function isStore(maybeStore: unknown): maybeStore is Store {
+  return (maybeStore as Store)._storeWrapper !== undefined;
+}
+
+function getWrapper(store: RecordDataStoreWrapper | Store): RecordDataStoreWrapper {
+  return isStore(store) ? store._storeWrapper : store;
+}
+
+export function peekGraph(store: RecordDataStoreWrapper | Store): Graph | undefined {
+  return Graphs.get(getWrapper(store));
+}
+
+export function graphFor(store: RecordDataStoreWrapper | Store): Graph {
+  const wrapper = getWrapper(store);
+  let graph = Graphs.get(wrapper);
+  if (graph === undefined) {
+    graph = new Graph(wrapper);
+    Graphs.set(wrapper, graph);
+  }
+  return graph;
+}
+
+export class Graph {
+  declare identifiers: Map<StableRecordIdentifier, Relationships>;
+  declare store: RecordDataStoreWrapper;
+  declare _queued: { belongsTo: any[]; hasMany: any[] };
+  declare _nextFlush: boolean;
+  declare implicitMap: Map<StableRecordIdentifier, RelationshipDict>;
+
+  constructor(store: RecordDataStoreWrapper) {
+    this.store = store;
+    this.identifiers = new Map();
+    this._nextFlush = false;
+    this._queued = { belongsTo: [], hasMany: [] };
```

Comment:
I do not believe anything accesses it. We haven't made use of private fields yet so I'd like to not start here but a cleanup pass should be done soon. Converting a whole lot of things to make use of private fields in one go will let us evaluate if there is a negative perf impact we might not otherwise perceive.